DonjinKrawler-Client/src/main/java/donjinkrawler/adapter/MediaAdapter.java

```diff
@@ -0,0 +1,36 @@
+package donjinkrawler.adapter;
+
+public class MediaAdapter implements MediaPlayer {
+    AdvancedMediaPlayer advancedMusicPlayer;
```

Comment:
Not sure if this is quite right. So you have an AdvancedMediaPlayer, which can play either wav or mp3, right? And the interface itself is implemented by WavPlayer and Mp3Player? And you have separate methods for playing wav and playing mp3. So what happens if I execute playmp3 on a WavPlayer? It doesn't really make sense to have two separate methods. I presume this is ridiculous behavior is because of the Adapter? I would rethink where to apply it if that's the case, or try to extend the functionality so it makes sense.